R/rows-dm.R

```diff
@@ -285,7 +285,19 @@ do_rows_append <- function(x, y, by = NULL, ..., in_place = FALSE, autoinc_col =
   source_rows <- map(key_values, ~ select(filter(y, !!returning == !!.x), -!!returning))
 
   con <- dbplyr::remote_con(x)
-  if (utils::packageVersion("dbplyr") >= "2.3.2.9000") {
+  # FIXME can be removed after depending on dbplyr >= 2.4.0
+  sql_query_append <- dbplyr::sql_query_append
+  append_args <- rlang::fn_fmls_names(dbplyr::sql_query_append)
+  old_interface <- identical(append_args, c("con", "x_name", "y", "...", "returning_cols"))
+  if (old_interface) {
+    target_name <- remote_name_qual(x)
+    insert_queries <- map(source_rows, ~ dbplyr::sql_query_append(
+      con,
+      target_name,
+      .x,
+      returning_cols = autoinc_col
+    ))
+  } else {
     dbplyr_ns <- asNamespace("dbplyr")
     remote_table <- mget("remote_table", dbplyr_ns, mode = "function", ifnotfound = list(NULL))[[1]]
```

Comment:
I can't quite follow what you mean. Are you worried about the use of `remote_table()`? This is a workaround because `remote_table()` isn't available in dbplyr < 2.4.0.
Previously, only `remote_name()` existed but the name didn't really make sense (_name_ implied a string to many people, but it also could have returned a schema or sql). See https://github.com/tidyverse/dbplyr/issues/1280 for more information.

Reply:
See above.